File: NotesAppRoomJava/app/src/main/java/com/example/notesapproom_java/MainActivity.java
package com.example.notesapproom_java;

import android.os.Bundle;

import com.example.notesapproom_java.database.NoteDatabase;
import com.example.notesapproom_java.repository.NoteRepository;
import com.example.notesapproom_java.viewmodel.NoteViewModel;
import com.example.notesapproom_java.viewmodel.NoteViewModelFactory;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.ViewModelProvider;
public class MainActivity extends AppCompatActivity {

    private NoteViewModel noteViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // Optional: Edge-to-edge support (not directly available like Kotlin DSL)
        // WindowCompat.setDecorFitsSystemWindows(getWindow(), false);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        setupViewModel();
    }

    private void setupViewModel() {
        NoteRepository noteRepository = new NoteRepository(NoteDatabase.getInstance(this));
        NoteViewModelFactory viewModelFactory = new NoteViewModelFactory(getApplication(), noteRepository);

        noteViewModel = new ViewModelProvider(this, viewModelFactory).get(NoteViewModel.class);
    }

    public NoteViewModel getNoteViewModel() {
        return noteViewModel;
    }
}

File: NotesAppRoom/app/src/main/java/com/example/notesapproom/MainActivity.kt
package com.example.notesapproom

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.lifecycle.ViewModelProvider
import com.example.notesapproom.database.NoteDatabase
import com.example.notesapproom.repository.NoteRepository
import com.example.notesapproom.viewmodel.NoteViewModel
import com.example.notesapproom.viewmodel.NoteViewModelFactory

class MainActivity : AppCompatActivity() {

    lateinit var noteViewModel: NoteViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)

        setupViewModel()
    }

    private fun setupViewModel(){
        val noteRepository = NoteRepository(NoteDatabase(this))

        val viewModelProviderFactory = NoteViewModelFactory(application, noteRepository)

        noteViewModel = ViewModelProvider(this, viewModelProviderFactory)[NoteViewModel::class.java]
    }
}